src/features/courses/course-controller.ts

```diff
@@ -875,20 +880,32 @@ class CourseController {
         } else {
             studentGrade = await workbook.getStudentGrade();
             if (studentGrade.courseWWTopicQuestionId !== options.questionId) {
-                throw new NotFoundError('The grade you have requested does not belong to the question provided');
+                throw new NotFoundError('The workbook you have requested does not belong to the question provided');
             }
         }
-        
 
-        if(_.isNil(workbook)) {
-            const workbooks = await studentGrade?.getWorkbooks({
-                limit: 1,
-                order: [ [ 'createdAt', 'DESC' ]]
-            });
-            workbook = workbooks?.[0] || null;
+        // if no workbookID, get the most recent workbook -- come back and delete this?
+        // if not enrolled, we don't even want to have workbooks
+        // if enrolled, workbooks are requested by ID for grades/statistics
+        // if enrolled, studentGrade holds currentProblemState
+        // if(_.isNil(workbook)) {
+        //     const workbooks = await studentGrade?.getWorkbooks({
+        //         limit: 1,
+        //         order: [ [ 'createdAt', 'DESC' ]]
+        //     });
+        //     workbook = workbooks?.[0] || null;
+        // }
+
+        // it may be undefined (user not enrolled)
+        // GetProblemParameters requires undefined over null
+        let formData: {[key: string]: unknown} | undefined = studentGrade?.currentProblemState;
+        // at this point, we only have a workbook if a valid workbookID was provided
+        // set the formData to match the contents of the workbook
+        if(!_.isNil(workbook)) {
+            formData = workbook.submitted.form_data;
         }
 
-
+        // studentGrade is the source of truth
```

Comment:
although in the future if a workbook is supplied it would be the source of truth
by the future I am talking about show me another or update randomization but you are looking at a previous attempt
so it would be:
```typescript
const randomSeed = workbook?.randomSeed ?? studentGrade?.randomSeed ?? null;
```